### PR TITLE
Added StarReLU Layer

### PR DESCRIPTION
**Pull request by mayank-root**

### CODE REVIEW

**Review by zoq — COMMENTED** [review]
Can you take a look at https://github.com/mlpack/mlpack/issues/3472 and https://github.com/mlpack/mlpack/pull/3478 add the `ActivationJacobianTest` for the activations functions?

File: src/mlpack/tests/ann/layer/star_relu.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  // Backward pass.
+  layer.Backward(output, gy, g);
+
+  // Test the Backward function.
+  REQUIRE(arma::approx_equal(g, delta, "absdiff", 1e-5));
+}
+
+/**
+ * JacobianTest for starRelu layer
```
Comment (zoq) [inline src/mlpack/tests/ann/layer/star_relu.cpp:67]:
```suggestion
 * JacobianTest for StarRelu layer
```

File: src/mlpack/tests/ann/layer/star_relu.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+#include "../../test_catch_tools.hpp"
+#include "../../catch.hpp"
+#include "../../serialization.hpp"
+#include "../ann_test_tools.hpp"
+
+using namespace mlpack;
+
+/**
+ * Simple test case for the staReLU layer.
```
Comment (zoq) [inline src/mlpack/tests/ann/layer/star_relu.cpp:24]:
```suggestion
 * Simple test case for the StaReLU layer.
```

File: src/mlpack/methods/ann/layer/star_relu.hpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+namespace mlpack {
+
+/**
+ * The StarReLU activation function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& s \cdot (\text{ReLU}(x))^2 + b
+ * f'(x) &=& s \cdot 2 \cdot \text{ReLU}(x)
+ * @f}
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/star_relu.hpp:27]: Can you cite the paper (https://arxiv.org/abs/2210.13452v2) as well, see https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/activation_functions/hard_swish_function.hpp for an example.

File: src/mlpack/methods/ann/layer/star_relu.hpp [hunk 4]
```diff
@@ -0,0 +1,119 @@
+/**
+ * @file methods/ann/layer/star_relu.hpp
+ * @author Mayank Raj
+ *
+ * Definition of StarReLU layer introduced byWeihao Yu, Chenyang Si, Pan Zhou,
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/star_relu.hpp:5]:
```suggestion
 * Definition of StarReLU layer introduced by Weihao Yu, Chenyang Si, Pan Zhou,
```

File: src/mlpack/tests/ann/layer/star_relu.cpp (hunk 1 above)
Comment (zoq) [reply src/mlpack/tests/ann/layer/star_relu.cpp:67]: Looks like we still have to change this.

**Comment by mayank-root** [comment]
@zoq Can you review this PR.

**Pull request closed without merging**